### ISSUE TITLE
HTTP Method Shiro Filter

### ISSUE THREAD

**Issue by cfollenf** [UKGovLD/registry-core#200]
A user trying to periodically fetch a resource from our registry submits kinds of preflight requests through HTTP HEADs. As they use those to control accessibility before submitting actual (valid) GETs, and [BasicAuthOnUpdateFilter](https://github.com/UKGovLD/registry-core/blob/master/src/main/java/com/epimorphics/registry/security/BasicAuthOnUpdateFilter.java) expectedly challenges them returning 401s, their whole retrieval process fails.

While HEADs are supposed to return identical bodiless responses, would it make sense to make this Shiro filter slightly more flexible so that it'd also allow HEAD requests through?

**Comment by der** [comment]
Sure, that would reasonable and easy to do.

**Comment by cfollenf** [comment]
As we remain unable to build a snapshot of the current registry-core .war ourselves could you publish a 2.3.21 release embedding such an update?

**Comment by der** [comment]
Hi @cfollenf. That should be done now. Thanks to @simonoakesepimorphics for doing the release. Let us know if there's any problems, or if you are happy with it then please close the issue.

**Comment by cfollenf** [comment]
Thanks so much to both of you for your reactivity, the fix has been successfully deployed, and conveniently works.
